### PR TITLE
test: add pytest-snapshot tests for TPC-H outputs

### PR DESCRIPTION
**Pull request by gforsyth**

### CODE REVIEW

File: ibis_substrait/tests/compiler/test_tpch.py [hunk 1]
```diff
@@ -704,8 +705,12 @@ def test_tpch1(tpc_h01, lineitem, compiler):
     "query",
     TPC_H,
 )
-def test_compile(query, compiler):
-    _ = compiler.compile(query)
+def test_compile(query, compiler, snapshot, request):
+    plan = compiler.compile(query)
+
+    snapshot.assert_match(
+        json_format.MessageToJson(plan), f"{request.node.callspec.id}.json"
```
Comment (jcrist) [inline ibis_substrait/tests/compiler/test_tpch.py:712]: Is the output of this guaranteed to be consistent (key ordering, ...)? It looks like there's a few [options we could set](https://googleapis.dev/python/protobuf/latest/google/protobuf/json_format.html#google.protobuf.json_format.MessageToJson) (`sort_keys=True` stands out), but if the output order is already consistent then that may not matter.

**Review by jcrist — APPROVED** [review]
Just a question, otherwise LGTM.

File: ibis_substrait/tests/compiler/test_tpch.py (hunk 1 above)
Comment (gforsyth) [reply ibis_substrait/tests/compiler/test_tpch.py:712]: I don't know if it's guaranteed or not, but as I've been working across different versions of substrait and Ibis, the field ordering matches the ordering in the proto definitions (as least for the defined fields).

I just ran it with `sort_keys=True` and it's hard to read, actually, after having spent long enough looking at the substrait plans in their "normal" form

**Comment by gforsyth** [comment]
Merging in b/c/ the branch protection rules are out-of-date (and were a tad premature)

**Pull request merged**